src/solver/conv_hip_implicit_gemm_f16f8f16_bwd_xdlops.cpp

```diff
@@ -0,0 +1,368 @@
+    }
+    else
+        return false;
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::IsValidValue() const
+{
+    return index < valid_kernels.size();
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::IsValid(
+    [[maybe_unused]] const ProblemDescription& problem) const
+{
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+    const auto& y_cast_type = problem.GetOut().GetCastType();
+    const auto& w_cast_type = problem.GetWeights().GetCastType();
+    if(y_cast_type == miopenBFloat8 && w_cast_type == miopenFloat8)
+        return CheckIsSupportCKArgs<ck::half_t, ck::bf8_t, ck::f8_t>(problem);
+#endif
+    return false;
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops& other) const
+{
+    return kernel_id == other.kernel_id;
+}
+
+PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops
+ConvHipImplicitGemmF16F8F16BwdXdlops::GetDefaultPerformanceConfig(
+    const ExecutionContext&, const ProblemDescription& problem) const
+{
+    PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops pp;
+    pp.HeuristicInit(problem);
+    return pp;
+}
+
+bool ConvHipImplicitGemmF16F8F16BwdXdlops::IsValidPerformanceConfig(
+    const ExecutionContext&,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops& config) const
+{
+    return config.IsValid(problem);
+}
+
+PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops
+ConvHipImplicitGemmF16F8F16BwdXdlops::Search(const ExecutionContext& ctx,
+                                             const ProblemDescription& problem,
+                                             const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemmF16F8F16BwdXdlops::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& ctx,
+    [[maybe_unused]] const ProblemDescription& problem) const
+{
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+    if(miopen::IsDisabled(MIOPEN_DEBUG_F16F8F16_CONV_IMPLICIT_GEMM_HIP_BWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(problem.HasMixedDataTypes())
+        return false;
+    if(!problem.IsTensorsCasted())
+        return false;
+    if(!problem.IsDirectionBackwardData())
+        return false;
+    if(!problem.Is3d())
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+    if(!problem.IsFp16())
+        return false;
+    if(!ck_utility::is_ck_whitelist(ctx.GetStream().GetDeviceName()))
+        return false;
+    const auto& yDesc = problem.GetOut();
+    const auto& wDesc = problem.GetWeights();
+    if(yDesc.GetCastType() && wDesc.GetCastType())
+    {
+        const auto y_cast_type = yDesc.GetCastType();
+        const auto w_cast_type = wDesc.GetCastType();
+        if(y_cast_type == miopenBFloat8 && w_cast_type == miopenFloat8)
```

Comment:
[R] Starting from C++17, std::optional may be directly compared with value, and this is enough:
```suggestion
    if(yDesc.GetCastType() == miopenBFloat8 && wDesc.GetCastType() == miopenFloat8)
```
Or, if you would like to use C++14 style, then compare like `*y_cast_type == miopenBFloat8`.

Reply:
Fixed it.